src/sentry/taskworker/processors/reply.py

```diff
@@ -0,0 +1,209 @@
+
+            # TODO this should read from task namespace configuration
+            deadletter_at = timezone.now() + timedelta(minutes=10)
+
+            return InflightActivation(
+                activation=activation,
+                status=TASK_ACTIVATION_STATUS_PENDING,
+                offset=offset,
+                added_at=Timestamp(seconds=int(time.time())),
+                deadletter_at=Timestamp(seconds=int(deadletter_at.timestamp())),
+                processing_deadline=None,
+            )
+
+        def limit_tasks(
+            message: Message[KafkaPayload],
+        ) -> KafkaPayload:
+            count = self.pending_task_store.count_pending_task()
+            if count >= self.max_inflight_activation_in_store:
+                # The number of pending inflight activations in the store exceeds the limit.
+                # Wait for workers to complete tasks before adding the next offset to the queue.
+                logger.info(
+                    "Number of inflight activation: %s exceeds the limit: %s. Retrying in 3 seconds",
+                    count,
+                    self.max_inflight_activation_in_store,
+                )
+                raise MessageRejected()
+            return message.payload
+
+        def do_upkeep(
+            message: Message[KafkaPayload],
+        ) -> KafkaPayload:
+            # Do upkeep tasks after processing replies.
+            # We should account for processing lag for replies here.
+            self.pending_task_store.handle_processing_deadlines()
+            self.pending_task_store.handle_retry_state_tasks()
+            self.pending_task_store.handle_deadletter_at()
+            self.pending_task_store.handle_failed_tasks()
+            self.pending_task_store.remove_completed()
+
+            return message.payload
+
+        def process_combined_message(message: Message[KafkaPayload]) -> None:
+            ((partition, _offset),) = message.committable.items()
+
+            # TODO smelling the topics is gross.
+            # We could use a message header instead?
+            topic = partition.topic
+            if topic.name == "hackweek":
+                limit_tasks(message)
+                activation = process_message(message)
+                logger.info("store activation for id=%s", activation.activation.id)
+
+                self.pending_task_store.store([activation])
+
+            if topic.name == "hackweek-reply":
+                result = ActivationResult()
+                result.ParseFromString(message.payload.value)
+                logger.info("process activation-result for id=%s", result.task_id)
+
+                self.pending_task_store.set_task_status(result.task_id, result.status)
+
+                # TODO only do once and a while, perhaps every 5s?
+                do_upkeep(message)
+
+        self.__send_thread = Thread(target=self.start_worker_push, daemon=True)
+        self.__send_thread.start()
+
```

Comment:
I think everytime a rebalance happens, `create_with_partitions` gets called and a thread that runs `start_worker_push` will start running. We probably need to send a termination signal to the thread that started in the previous call to `create_with_partitions` before we start this one.